typing/typecore.ml

```diff
@@ -1301,7 +1320,7 @@ let add_module_variables env module_variables =
   ) env module_variables_as_list
 
 let enter_variable ?(is_module=false) ?(is_as_variable=false) tps loc name mode
-    ty attrs =
+    ?(kind = Val_reg) ty attrs =
```

Comment:
I think in general optional parameters like this are pretty error prone and best avoided unless there's a good specific reason for them.  This function only has a few call sites, so I propose we just specify this at all of them.